docs/dg/dev/architecture/dependency-injection.md

```diff
@@ -0,0 +1,257 @@
+     * filtering out modules that are not relevant for the current environment.
+     *
+     * F.e. In a dev environment you want to include all modules (composer:require-dev is installed), while in a production
+     * environment you don't have all modules available but code in your project that extends it for such cases the compilation
+     * process would fail if those modules are not found.
+     */
+    $moduleFinder = new ModuleFinderFacade();
+    $projectModules = $moduleFinder->getProjectModules();
+
+    foreach ($projectModules as $moduleTransfer) {
+        /**
+         * Skip excluded modules entirely when configured in the `$excludedModuleConfiguration`.
+         */
+        if (isset($excludedModuleConfiguration[$moduleTransfer->getName()]) && !is_array($excludedModuleConfiguration[$moduleTransfer->getName()])) {
+            continue;
+        }
+
+        // Organization may be Pyz, or any of on the project level defined ones
+        $organization = $moduleTransfer->getOrganization()->getName();
+
+        foreach ($moduleTransfer->getApplications() as $applicationTransfer) {
+            // If you want to exclude specific applications within a module, you can skip them here.
+            //
+            // Example:
+            //
+            // if ($applicationTransfer->getApplication() === 'UnwantedApplication') {
+            //    continue;
+            // }
+
+            $namespace = sprintf('%s\\%s\\', $organization, $applicationTransfer->getName());
+
+            /**
+             * Here is the path built to the services directory of the module for the specific application.
+             * 
+             * This path structure is based on the standard Spryker project structure.
+             */
+            $path = sprintf(
+                '../../../src/%s/%s/%s/',
+                $organization,
+                $applicationTransfer->getName(),
+                $moduleTransfer->getName(),
+            );
+
+            /**
+             * This is the important part: Load all services from the module's application services directory into the DIC.
+             * 
+             * You can also only use this line instead of the whole code in this file. But then you would have to make sure
+             * that all modules you want to be available in the DIC are included manually here, which is not the preferred way.
+             */
+            $services->load($namespace, $path);
+        }
+    }
+};
+
+```
+
+### Autowiring
+
+You can use autowiring to automatically inject dependencies into your services. For details, see [Autowiring](https://symfony.com/doc/current/service_container/autowiring.html) in the Symfony documentation.
+
+Since Spryker's core modules are not yet using the Symfony Dependency Injection component, you can't autowire classes from the core. You need to manually add them to your service configuration if you want to inject core dependencies.
```

Comment:
OPTIONAL: Can we be more proactive and add example here on how to do it, to simplify that task for project devs?